tests/Unit/CommandLine/Library/EnvironmentTest.php

```diff
@@ -128,12 +128,27 @@ public function isLocalBranchEqualToCachesLocalHeadHash()
         $mockedLocalCommitHash = '0ewq321';
 
         $this->mockedProcessRunner->shouldReceive('runAsProcess')->once()
-            ->with('git rev-list -n 1 "HEAD"')->andReturn($mockedLocalCommitHash);
+            ->with('git rev-list -n 1 \'HEAD\'')->andReturn($mockedLocalCommitHash);
         $this->mockedProcessRunner->shouldReceive('runAsProcess')->twice()
-            ->with('git rev-list -n 1 "' . $mockedBranchName . '"')->andReturn($mockedCommitHash);
+            ->with('git rev-list -n 1 \'' . $mockedBranchName . '\'')->andReturn($mockedCommitHash);
 
-        $result = $this->subject->isLocalBranchEqualTo($mockedBranchName);
+        $this->subject->isLocalBranchEqualTo($mockedBranchName);
         $result = $this->subject->isLocalBranchEqualTo($mockedBranchName);
         self::assertFalse($result);
     }
+
+    /**
+     * @test
+     */
+    public function commitishToHashIsInjectionResistent()
+    {
+        $mockedBranchName = 'HEAD; killall php;';
```

Comment:
der name bricht nicht aus den single quotes aus oder hab' ich noch zu wenig kaffee intus? eher  'HEAD\' && killall php'

Reply:
Das Escaping wird getestet. Gäbe es kein Escaping würde dieser "Branchname" brechen. Ich vertraue etwas darauf, dass escapeshellarg tut was es soll ^^